ai/STA/Strategy/defense_wall.py

```diff
@@ -27,14 +27,14 @@ def __init__(self, game_state: GameState, number_of_players: int = 4):
                              Role.FIRST_DEFENCE, Role.SECOND_DEFENCE]
 
         goalkeeper = self.game_state.get_player_by_role(Role.GOALKEEPER)
-        self.add_tactic(Role.GOALKEEPER, GoalKeeper(self.game_state, goalkeeper, ourgoal))
+        self.create_node(Role.GOALKEEPER, GoalKeeper(self.game_state, goalkeeper, ourgoal))
 
         role_by_robots = [(i, self.game_state.get_player_by_role(i)) for i in roles_to_consider]
         self.robots = [player for _, player in role_by_robots if player is not None]
         for role, player in role_by_robots:
             if player:
-                self.add_tactic(role, AlignToDefenseWall(self.game_state, player, self.robots))
-                self.add_tactic(role, GoKick(self.game_state, player, target=self.theirgoal))
+                self.create_node(role, AlignToDefenseWall(self.game_state, player, self.robots))
+                self.create_node(role, GoKick(self.game_state, player, target=self.theirgoal))
 
                 self.add_condition(role, 0, 1, partial(self.is_closest, player))
                 self.add_condition(role, 1, 1, partial(self.is_closest, player))
```

Comment:
You need to upgrade those add_condition to fit the connect_to() model we now have.